### ISSUE TITLE
Add Ashmem and binder support to the Surface kernel

### ISSUE THREAD

**Issue by SoftwareRat** [linux-surface/linux-surface#761]
Hello,
Linux on the Surface is awesome, thanks to the work. What would be cool, is to run Android apps on it.
There is [Waydroid](https://waydro.id) but it requires Ashmem and binder. 

Can you please build kernel with Ashmem and Binder support, specially for [Fedora](https://yanqiyu.info/2021/10/25/waydroid-fedora/)

### Environment
- Hardware model: Surface Go 2 (m3-8100Y, 8GB)
- Kernel version: `Linux srt-surface-fedora 5.16.11-1.surface.fc35.x86_64 #1 SMP PREEMPT Thu Feb 24 14:52:52 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux`
- Distribution: Fedora

**Comment by StollD** [comment]
Binder is already enabled in the default Fedora kernel, and should therefor also be enabled in our builds. Ashmem doesn't seem to be required anymore (haven't tested it yet), see https://copr.fedorainfracloud.org/coprs/aleasto/waydroid

**Comment by 110errm** [comment]
Hello.
What about arch kernel? Is binder or ashmem enabled? Inserting as a module will end up in system goin down. Is it possible to integrate ashmem an binder or do i have to compile a kernel by myself?

Surface Pro 2017 EndeavourOS 5.18.12-arch1-1-surface

Thank u for a reply

**Comment by qzed** [comment]
Ashmem got removed upstream. See e.g. https://github.com/anbox/anbox/issues/2042. So even if you wanted to compile a kernel yourself, you'd need to revert that upstream change first. I'd recommend figuring out why building it as a module doesn't work or looking for replacements (which should be memfd, as mentioned in the commit message removing it: https://lore.kernel.org/all/20220315123457.2354812-1-hch@lst.de/).

**Comment by gl0zzy** [comment]
Can the current linux-surface arch kernel use waydroid?
It requires the kernel to include binder.

**Comment by StollD** [comment]
No, it can't. We follow the default Arch kernel configuration, which disables binder.

**Comment by gl0zzy** [comment]
Actually I don't know much about linux kernel or binder, but I tried running waydroid and it works fine.
Thank you for creating this great project.
